apps/c_backend/pipeline_cpp.cpp

```diff
@@ -0,0 +1,84 @@
+#include "Halide.h"
+
+using namespace Halide;
+using Internal::Call;
+
+Expr make_call_cpp_extern_toplevel(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "cpp_extern_toplevel", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+
+Expr make_call_cpp_extern_namespace(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace1::cpp_extern", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+
+Expr make_call_cpp_extern_shared_namespace_1(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace2::cpp_extern_1", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+Expr make_call_cpp_extern_shared_namespace_2(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace2::cpp_extern_2", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+Expr make_call_cpp_extern_shared_namespace_3(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace2::cpp_extern_3", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+
+Expr make_call_cpp_extern_nested_namespace_outer(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace_outer::cpp_extern", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+Expr make_call_cpp_extern_nested_namespace_inner(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace_outer::namespace_inner::cpp_extern", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+
+Expr make_call_cpp_extern_shared_nested_1(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace_shared_outer::cpp_extern_1", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+Expr make_call_cpp_extern_shared_nested_2(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace_shared_outer::cpp_extern_2", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+Expr make_call_cpp_extern_shared_nested_3(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace_shared_outer::inner::cpp_extern_1", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+Expr make_call_cpp_extern_shared_nested_4(Expr arg1, Expr arg2) {
+    return Call::make(type_of<int>(), "namespace_shared_outer::inner::cpp_extern_2", {cast<int>(arg1), cast<float>(arg2)}, Call::ExternCPlusPlus);
+}
+
+// Make sure extern "C" works
+HalideExtern_2(int, an_extern_c_func, int, float);
+
+int main(int argc, char **argv) {
+    Func f;
+    ImageParam input(UInt(16), 2);
+    Var x, y;
+
+    Expr add_all_the_things = cast<int>(0);
```

Comment:
This name left me hoping for a Hyperbole And A Half shout-out. Left disappointed.